.github/workflows/compose.yml

```diff
@@ -0,0 +1,57 @@
+name: Docker Compose deployment 
+
+on:
+  pull_request:
+    branches:
+      - master
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  docker-compose:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+
+    - name: Log into Docker registry
+      env:
+        DOCKER_REGISTRY_URL: tip-tip-wlan-cloud-ucentral.jfrog.io
+        DOCKER_REGISTRY_USERNAME: ucentral
+      uses: docker/login-action@v1
+      with:
+        registry: ${{ env.DOCKER_REGISTRY_URL }}
+        username: ${{ env.DOCKER_REGISTRY_USERNAME }}
+        password: ${{ secrets.DOCKER_REGISTRY_PASSWORD }}
+
+    - name: Create certificate and key files from Github secrets
+      env: 
+        DIGICERT_TIP_WILDCARD_CERT: ${{ secrets.DIGICERT_TIP_WILDCARD_CERT }}
+        DIGICERT_TIP_WILDCARD_KEY: ${{ secrets.DIGICERT_TIP_WILDCARD_KEY }}
+      run: |
+        cp certificates/root.pem docker-compose/ucentral-data/certs/restapi-ca.pem
+        'echo "$DIGICERT_TIP_WILDCARD_CERT" > docker-compose/ucentral-data/certs/{websocket-cert.pem,restapi-cert.pem}'
+        'echo "$DIGICERT_TIP_WILDCARD_KEY" > docker-compose/ucentral-data/certs/{websocket-key.pem,restapi-key.pem}'
+
+    - name: Instantiate Docker Compose deployment
+      env:
+        DOCKER_COMPOSE_PROJECT_NAME: ucentral-${{ github.sha }}
+        UCENTRALGW_TAG: ${{ github.sha }}      
+      run: docker-compose -f docker-compose/docker-compose.yml --project-directory docker-compose/ up -d
+
+    - name: Check functionality of uCentralGW
+      env: 
+        UCENTRALGW: "${{ github.sha }}.cicd.lab.wlan.tip.build:16001" 
+        FLAGS: "-s --resolve $UCENTRALGW:127.0.0.1"
+      run: |
+        apt-get install jq
+        ./test_scripts/curl/cli listdevices 
+
+    - name: Re-tag image if test was successful and PR was merged
```

Comment:
I don't think retagging a PR image as master is the right way. What if the master does include new changes which will then be overridden? IMO we should simply trigger a build on each push to master. This way we can make sure that the master Docker image is always equivalent to master branch.

Reply:
You're right, didn't think about PRs which haven't been rebased before. In this case this doesn't make sense at this point.